src/portable/portable.c

```diff
@@ -1005,18 +1005,26 @@ static int install_chroot_dropin(
                                "LogExtraFields=PORTABLE=", base_name, "\n"))
                         return -ENOMEM;
 
+                if (!ordered_hashmap_isempty(extension_images))
+                        if (!strextend(&text, "LogExtraFields=PORTABLE_ROOT=", image_path, "\n"))
```

Comment:
why did you name this "_ROOT="? Why not PORTABLE_PATH= or so?

Also, shouldn't this be: `m->image_path ?: image_path`?

Reply:
i cannot parse that? how is that used for RootImage=?

I wuld assume ROOT= would actually refer to a dir, not an image, hence i am irritated by this.

what's the usecase for this btw? why isn't the basename of the image enough, i.e. the current PORTABLE= field? why do you need the path too?

Reply:
Maybe an example would help to clarify - this is attached with 2 extensions and a base, and the unit file is extracted from the app0 extension, 

```
# Drop-in created for image '/home/bluca/git/systemd/base.raw:/home/bluca/git/systemd/app0.raw:/home/bluca/git/systemd/app1.raw', do not edit.

[Service]
RootImage=/home/bluca/git/systemd/base.raw
Environment=PORTABLE=app0.raw
BindReadOnlyPaths=/etc/os-release:/run/host/os-release
LogExtraFields=PORTABLE=app0.raw
LogExtraFields=VERSION_ID=10
LogExtraFields=ID=debian
LogExtraFields=PORTABLE_ROOT=/home/bluca/git/systemd/base.raw

ExtensionImages=/home/bluca/git/systemd/app0.raw
LogExtraFields=PORTABLE_EXTENSION=/home/bluca/git/systemd/app0.raw
LogExtraFields=SYSEXT_VERSION_ID=app0:0

ExtensionImages=/home/bluca/git/systemd/app1.raw
LogExtraFields=PORTABLE_EXTENSION=/home/bluca/git/systemd/app1.raw
LogExtraFields=SYSEXT_VERSION_ID=app1:1
```

- PORTABLE= is the image _name_ of the source of the unit file, so in this case it's `app0.raw` as that's where the unit is from
- PORTABLE_ROOT points to the base image, which is used as RootImage= or RootDirectory= - the distinction is not important for this purpose, the important thing is what is used as a base, the name itself encodes whether it's a directory or an image (e.g. `base/` vs `base.raw`)
- PORTABLE_EXTENSION points to each extension image/directory, again used as ExtensionImages= or ExtensionDirectories= same as above

The use case is that it's useful to me to know when looking retrospectively at old, archived log messages for the unit, exactly on which image they are running on

Reply:
in the interest of saving time I've switched to using the basename everywhere rather than the full path, does this work better now? Example:

```
[Service]
RootImage=/home/bluca/git/systemd/base.raw
Environment=PORTABLE=app0.raw
BindReadOnlyPaths=/etc/os-release:/run/host/os-release
LogExtraFields=PORTABLE=app0.raw
LogExtraFields=VERSION_ID=10
LogExtraFields=ID=debian
LogExtraFields=PORTABLE_ROOT=base.raw

ExtensionImages=/home/bluca/git/systemd/app0.raw
LogExtraFields=PORTABLE_EXTENSION=app0.raw
LogExtraFields=SYSEXT_VERSION_ID=app0:0

ExtensionImages=/home/bluca/git/systemd/app1.raw
LogExtraFields=PORTABLE_EXTENSION=app1.raw
LogExtraFields=SYSEXT_VERSION_ID=app1:1
```